src/csharp/Microsoft.Spark/ML/Feature/IDF.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+using Microsoft.Spark.Sql;
+
+namespace Microsoft.Spark.ML.Feature
+{
+    /// <summary>
+    /// Inverse document frequency (IDF). The standard formulation is used:
+    /// idf = log((m + 1) / (d(t) + 1)), where m is the total number of documents and d(t) is
+    /// the number of documents that contain term t.
+    /// 
+    /// This implementation supports filtering out terms which do not appear in a minimum number
+    /// of documents (controlled by the variable minDocFreq). For terms that are not in at least
+    /// minDocFreq documents, the IDF is found as 0, resulting in TF-IDFs of 0.
+    /// </summary>
+    public class IDF : IJvmObjectReferenceProvider
+    {
+        private static readonly string s_IDFClassName = "org.apache.spark.ml.feature.IDF";
+        
+        /// <summary>
+        /// Create a <see cref="IDF"/> without any parameters
+        /// </summary>
+        public IDF()
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(s_IDFClassName);
+        }
+
+        /// <summary>
+        /// Create a <see cref="IDF"/> with a UID that is used to give the
+        /// <see cref="IDF"/> a unique ID
+        /// </summary>
+        /// <param name="uid">An immutable unique ID for the object and its derivatives.</param>
+        public IDF(string uid)
+        {
+            _jvmObject = SparkEnvironment.JvmBridge.CallConstructor(s_IDFClassName, uid);
+        }
+        
+        internal IDF(JvmObjectReference jvmObject)
+        {
+            _jvmObject = jvmObject;
+        }
+
+        private readonly JvmObjectReference _jvmObject;
```

Comment:
Can we make this consistent with existing code? I think we declare private variables first. Check https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Sql/DataFrameWriter.cs for example.